### PR TITLE
github: connectionId -> connectorId

### PR DESCRIPTION
**Pull request by spolu**
## Description

- All activities: accept `connectionIdOrConnectorId: ConnectionIdOrConnectorId = ModelId | string;` and based on the type retrieve the connector from the `connectionId` or the `connectorId`. Support currently running workflows starting activities with `connectionId: string` and newly created workflows which will start activities with `connectorId: ModelId`. This is safe since there is no change of number of args.
- All workflows transition to using `connectorId` everywhere
- Clients updated
- All other remain unchanged since the activities are in charge of fetching the `connector: ConnectorResource` and then call in the sub function with the `connectionId` as before.

## Risk

High for github, low overall.

## Deploy Plan

- deploy `connector`
- once all old workflows are done, remove `connectionIdOrConnectorId` and remove the `connectionId` param in `findByDataSourceAndConnection` as well as the one introduced here: `fetchByConnectionId`

### CODE REVIEW

File: connectors/src/connectors/github/temporal/workflows.ts [hunk 1]
```diff
@@ -147,7 +144,6 @@ export async function githubReposSyncWorkflow(
             {
               dataSourceConfig,
               connectorId,
-              connectionId,
```
Comment (spolu) [inline connectors/src/connectors/github/temporal/workflows.ts:150]: @fontanierh this one is the only one where I have a doubt. Maybe I'll keep connectionId here just to be safe

File: connectors/src/connectors/github/temporal/client.ts [hunk 2]
```diff
@@ -67,13 +65,7 @@ export async function launchGithubFullSyncWorkflow({
...
 
   await client.workflow.start(githubFullSyncWorkflow, {
-    args: [
-      dataSourceConfig,
-      connectionId,
-      connectorId,
-      syncCodeOnly,
-      forceCodeResync,
-    ],
+    args: [dataSourceConfig, connectorId, syncCodeOnly, forceCodeResync],
```
Comment (fontanierh) [inline connectors/src/connectors/github/temporal/client.ts:68]: So we're changing the number of params ? How will this work ?

File: connectors/src/connectors/github/temporal/workflows.ts (hunk 1 above)
Comment (fontanierh) [reply connectors/src/connectors/github/temporal/workflows.ts:150]: Yeah this won't be backwards compatible I think

**Review by fontanierh — COMMENTED** [review]
Not sure I fully understand how we can just change workflow positional params without making a separate workflow / queue ?

**Review by fontanierh — APPROVED** [review]

File: connectors/src/connectors/github/temporal/client.ts (hunk 2 above)
Comment (spolu) [reply connectors/src/connectors/github/temporal/client.ts:68]: This will fail I'll cancel :+1:

**Comment by spolu** [comment]
Actually went for a direct cut (forget instructions in the PR description).
This will fail some workflows likely 1 or 2 that we will restat manually from the machines

Will test a fullSync from development before merging

**Pull request merged**